File: 0011-container-with-most-water/solution.c
int maxArea(int* h, int n) {
    int maxArea=0;
    int left=0,right=n-1;
    while(left<right){
        int minArea = fmin(h[left],h[right]);
        int curArea = minArea * (right-left);
        maxArea = fmax(curArea,maxArea);
        printf("%d\n",maxArea);
        if(h[left]<h[right]){
            left++;
        }
        else{
            right--;
        }
    }
    return maxArea;
}

File: 0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] h) {
        int area = 0;
        int n = h.length;
        int left = 0;
        int right = n - 1;
        while(left < right){
            int height = Math.min(h[left],h[right]);
            int width = right - left;
            area = Math.max(area,height*width);
            if(h[left] < h[right]){
                left++;
            }else{
                right--;
            }
        }
        return area;
    }
}
